torchao/csrc/cuda/mx_kernels/mx_fp_cutlass_kernels.cu

```diff
@@ -3,6 +3,11 @@
 //
 // This source code is licensed under the BSD 3-Clause license found in the
 // LICENSE file in the root directory of this source tree.
+
+// Ensure this file is only compiled with sm100a architecture
+#if !defined(__CUDA_ARCH__) || (__CUDA_ARCH__ < 1000)
```

Comment:
@drisspg Have you tested this PR on Blackwell?

How I'm building this on my 5090 (sm120)
- set `build_for_sm100a = True` in `setup.py` to compile this file
- set `TORCH_CUDA_ARCH_LIST=12.0a`

I face compile error at this line. I believe it's because `__CUDA_ARCH__` is not defined in host code, hence this file can't be compiled at all.

Reply:
this is why you dont vibe code

Reply:
https://github.com/pytorch/ao/pull/2284